portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java

```diff
@@ -37,29 +39,34 @@
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         long id = Long.parseLong(request.getParameter("id"));
 
-        //get comment entity corresponding to id from datastore
-        Key commentEntityKey = KeyFactory.createKey("Comment", id);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Transaction txn = datastore.beginTransaction();
 
         try {
```

Comment:
You need to use the txn methods to fetch and put data. For an example, see https://cloud.google.com/datastore/docs/concepts/transactions

Note that they use `txn.fetch` and `txn.put` to fetch and store data as opposed to `datastore.get` and `datastore.put`. This ensures those operations are done as part of the transaction.

Reply:
I'm not sure what `fromKey` and `toKey` are in this example. I was following the Java 8 example here: https://cloud.google.com/appengine/docs/standard/java/datastore/transactions and they use `datastore.get `and `datastore.put`.

Reply:
Oh, I see we're using the old API. In this case, you still need to call `datastore.put` with `txn` as the first argument.